internal/serviceapi/orgserviceaccountsecretapi/custom_operations.go

```diff
@@ -0,0 +1,47 @@
+package orgserviceaccountsecretapi
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/autogen"
+)
+
+type response struct {
+	Secrets []map[string]any `tfsdk:"secrets"`
+}
+
+func (r *rs) PerformRead(ctx context.Context, req *autogen.HandleReadReq) ([]byte, *http.Response, error) {
+	model := req.State.(*TFModel)
+
+	bodyResp, apiResp, err := autogen.CallAPIWithoutBody(ctx, req.Client, req.CallParams)
+	if err != nil {
+		tflog.Error(ctx, "Failed to read service account secret", map[string]any{
+			"error": err.Error(),
+		})
+		return nil, apiResp, err
+	}
+	if autogen.NotFound(bodyResp, apiResp) {
+		return bodyResp, apiResp, nil
+	}
```

Comment:
if SA is not found, I assume `err` is set, right?
If this is tru, then the notfound block should be nested inside the err handling
```
	if err != nil {
	    if autogen.NotFound(bodyResp, apiResp) {
		    return bodyResp, apiResp, nil // I think we should return the err here
	    }
		tflog.Error(ctx, "Failed to read service account secret", map[string]any{
			"error": err.Error(),
		})
		return nil, apiResp, err
	}
```